panel/applets/icon-tasklist/WindowManager.vala

```diff
@@ -0,0 +1,19 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright (C) 2016 Fernando Mussel <fernandomussel91@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+public interface WindowManager
```

Comment:
One extra thing regarding the WindowManager. Although the interface might be unnecessary, I am in favor of keeping at least some of the functionality it exposes concentrated in IconTasklistApplet to avoid code duplication and keeping things more organized. For instance:
 * *toggle_window*: the function is a bit convoluted and is used in three different places; two times in Button.vala and once in WindowListItem.vala

* *toggle_window_list*: used only once in Button.vala, but it requires a reference to popover_manager. We could give a reference to the manager to every button, but IconTasklistApplet already has to listen to changes to the popover_manager and unregister the popovers belonging to applications that are closed completely (0 windows left). So I thought it would be better to just let IconTasklistApplet handle display the popover.
* *close_window*: used in only two places. It is simple enough that we could probably just remove it.
* *query_window*: I created it to avoid having to pass yet another object, in this case an AppSystem instance, to Application. I don't know. Maybe we should keep this function.